WinInstaller/WinInstaller/fsutils.cpp

```diff
@@ -0,0 +1,49 @@
+#include "fsutils.h"
+#include "winapi.h"
+
+
+namespace mywininstaller
+{
+	namespace fsutils
+	{
+		using std::filesystem::path;
+
+
+		void createDirectoryAndParentsTransacted(Transaction<path>& transaction, const path& dirPath)
+		{
+			const path& root = dirPath.root_path();
+			path subpath = "";
+			bool biggerThanRoot = false;
+			for (const path& part : dirPath)
+			{
+				subpath /= part;
+				if (biggerThanRoot || subpath > root)
```

Comment:
why even perform this check each iteration? just skip the root in advance, then begin the loop